feat(vpcv2): implementation of add gateway method

shikha372 opened the pull request:
### Issue # (if applicable)

Tracking  #[30762](https://github.com/aws/aws-cdk/issues/30762).

### Reason for this change

implementing below methods for vpcV2.
`routeTable.addroute(destination, target)`:   Adds a new route to the existing route table of the subnet.

`vpc.enableVpnGatewayV2()`:  added a new function for the customer to add VPNGateway to their VPC. In the options, user can specify list of subnets for VPNRoutePropogation. This is similar to previous implementation, only difference is with VPNGateway L2, it is now creating VPNGatewayV2 which implements IRouteTarget and hence can be used a destination to be set up in route tables.

`addInternetGateway` : adds internetGW to the VPC. 
**Default behaviour:**  add default route with destination set to ‘0.0.0.0’ and ‘::0’(in case of subnet with ipv6). Also a check in place to verify SubnetType is set to public as IGW is meant to be added to public subnets.

`addNatGateway`:  NatGateways are subnet specific and are usually associated with PRIVATE_WITH_EGRESS or PUBLIC subnet. Also, one can’t attach NGW(Public) to subnet if VPC doesn’t have an IGW attached to it. This is validated in method implementation to prevent runtime deployment error.

**No default behaviour** for the routes, it takes in the single subnet option and associates a NATGW with it. 

`vpc.addEgressOnlyInternetGateway()`: Egress Only internet GW are meant for outbound ipv6 traffic which can be custom or all ipv6(::/0).

**Default behaviour:** Associates a EIGW to the vpc and takes optional input for subnets to define a default route in associated route Table, if a destination is not provided, then it is defined as all outbound ipv6 in subnet’s route table.

**Additional changes:**
-> Modify Readme
-> Separate ipam related Tests

### Use Case
<!--What is the bug or use case behind this change?-->

Allows user to define gateways in their vpc with a simple method and an optional default route setup on provided subnets.

<!--What code changes did you make? Have you made any important design decisions?-->
Note: Breaking change since previously VPNGateway was released under route class,  we’ve modified it to VPNGatewayV2.
`vpc.enableVpnGateway` is marked as deprecated in vpcv2 base class.

### Description of how you validated changes

Added unit tests and integration tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: f42666d73add8c0fc93b49078667d26c44ac4c06
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=41056a6d-8f8a-4e20-ad48-3667c197b729%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 36a9511375d24d15b20b1e5b5779f838a0eabc91
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a1f0c26a-e42a-43a9-bb90-b5030f21f2fe%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 9ecdb496d387cf4f190765604ccfb387649467b8
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=caf22d64-c667-43ec-9a85-90801bf78057%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 5bd279bb2eab38540c30aa691362b0a345d66408
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=20c440ac-54c7-4a35-852d-c9c4460a3e3e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 6023ecdf8e8415fc3f1795a9047a874a0c168db2
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c38591de-ba71-44c7-8b84-c5248e87192a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 85eafa29ac355155ef5808d97dc9c0ca91e6648a
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=af486c06-d743-4494-ba7c-a3ed1d4869a9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 02c3ff919f36b0fc618b05788690db4ea565d95f
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a79cd9a6-355f-437b-8581-5579bd99f9bf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 78ec5ce9ce8fe092562446781b8568e86abb6b6f
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=43dedf20-3fab-4f8d-87d6-ebd77bee2ebb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 5a05757fc8260b68c4141d61d4ae7cc43adb4843
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=321441ab-bffd-4b56-86df-91a918eacdb4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 391ae8e0d6ba28e58a9258c76943257d9ec6168c
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5ede028f-6b08-4b92-9764-66cb133b309c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 1666c572b4c983c5a19630fbf64c729cfd6940b1
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=07a642e9-2534-4047-8c26-822c22ba8fc5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d9e453feb14cca719a78210b6b77ea938a7f3549
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=28f95484-c577-479c-89cb-759e2badc5b3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 475ec3a1de8b5ef1bb0298816d632c80ecd3d65f
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=849494c8-ce5e-45b9-b6c1-85e7d210defc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 98c6ee7fb6bf42a3b5e60b9e9bd718ee9e0f4f16
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e0ac05a5-2943-4d6f-a58e-8d6e675b340c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: aba5e6619016b8938bedf954ad77d5a3679339ca
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0ec4d3f9-8d01-4398-ae99-b1d919683645%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: aeba62bbe7cc9b08eecda5f9f9a5f65a48ac33e8
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=826c67f0-4ffa-412a-aae3-a6f7a5f50d60%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 2f1e60c14ed4552cfe59d80993bafd8a5d4accbb
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9a517465-65f3-4480-8bfc-5e66f8fdf7ac%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: e0602781b143f9758a65f83e0c0bf26669c87d9f
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7973055e-de2a-472a-a5b2-0b05811a47e0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 74adb43731d0b17d0069b5296b68da4b424a15f3
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8713c2b5-8dbf-4f7b-a121-ea414d34f65a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d136789368d355cc136baf12cb9fdc67e243d056
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=92a352d5-20f7-4a51-8680-73ba9ba08b7e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -569,6 +617,23 @@ export class RouteTable extends Resource implements IRouteTable, IDependable {
   }
+
+  /**
+   * Adds a new custom route to the route table.
+   * @param destination The IPv4 or IPv6 CIDR block used for the destination match.
+   * @param target The gateway or endpoint targeted by the route.
```
```suggestion
   * Adds a new custom route to the route table.
   *
   * @param destination The IPv4 or IPv6 CIDR block used for the destination match.
   * @param target The gateway or endpoint targeted by the route.
```

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -286,7 +291,30 @@ export class VPNGateway extends Resource implements IRouteTarget {
+
+    // Propagate routes on route tables associated with the right subnets
+    const vpnRoutePropagation = props.vpnRoutePropagation ?? [{}];
+    const routeTableIds = allRouteTableIds(flatten(vpnRoutePropagation.map(s => props.vpc.selectSubnets(s).subnets)));
```
What is this line doing?

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -286,7 +291,30 @@ export class VPNGateway extends Resource implements IRouteTarget {
+    // The AWS::EC2::VPNGatewayRoutePropagation resource cannot use the VPN gateway
+    // until it has successfully attached to the VPC.
+    // See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpn-gatewayrouteprop.html
+    this._routePropagation.node.addDependency(this._attachment);
+
   }
```
```suggestion
    this._routePropagation.node.addDependency(this._attachment);
  }
```

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -331,6 +359,12 @@ export class NatGateway extends Resource implements IRouteTarget {
 
+    if (this.connectivityType == NatConnectivityType.PUBLIC) {
+      if (!props.vpc && !props.allocationId) {
+        throw new Error('Either provide vpc or allocationId');
```
Is providing both acceptable?

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -491,39 +525,53 @@ export class Route extends Resource implements IRouteV2 {
+    this.destination = props.destination;
+    const isDestinationIpv4 = NetworkUtils.validIp(props.destination);
+    if (!isDestinationIpv4) {
+      //TODO Validate for IPv6 CIDR range
```
is this intentionally left to be addressed later on?

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -569,6 +617,23 @@ export class RouteTable extends Resource implements IRouteTable, IDependable {
+   * @param target The gateway or endpoint targeted by the route.
+   */
+  public addRoute(id: string, destination: string, target: RouteTargetType) {
+    if (!target.gateway && !target.endpoint) {
+      throw new Error('Target is defined without a gateway or endpoint.');
+    }
```
I feel like this check is not necessary if we tweak the implementation of `RouteTargetType` to enforce that exactly one of the two types are specified in the constructor. Something like this:

```ts
    if ((gateway && endpoint) || (!gateway && !endpoint)) {
      throw new Error("Exactly one of 'gateway' or 'endpoint' must be specified.");
    }

    if (gateway) {
      this.gateway = gateway;
    } else {
      this.endpoint = endpoint;
    }
```

shikha372 replied:
good point, validation can be in `RouteTarget`, moved it in latest rev.

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/subnet-v2.ts:
```diff
@@ -95,6 +96,12 @@ export interface ISubnetV2 extends ISubnet {
   readonly ipv6CidrBlock?: string;
 
+  /**
+   * The type of subnet (public or private) that this subnet represents.
+   * @attribute SubnetType
```
```suggestion
   * The type of subnet (public or private) that this subnet represents.
   *
   * @attribute SubnetType
```

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -1,7 +1,73 @@
+ */
+export interface EgressOnlyInternetGatewayOptions{
+  /**
+   * List of subnets where route to EGW will be added
+   * @default no route created
```
```suggestion
   * List of subnets where route to EGW will be added
   *
   * @default - no route created
```

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -1,7 +1,73 @@
+  readonly subnets?: SubnetSelection[];
+
+  /**
+   * Destination Ipv6 address for EGW route
+   * @default '::/0' all Ipv6 traffic
```
```suggestion
   * Destination Ipv6 address for EGW route
   *
   * @default - '::/0' all Ipv6 traffic
```
Can you clarify what this default value means? It feels a bit confusing when I read it.

shikha372 replied:
default means setting up outbound to all IPv6 address for which the range is '::/0'

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -1,7 +1,73 @@
+export interface InternetGatewayOptions{
+
+  /**
+   * Destination Ipv6 address for EGW route
+   * @default '0.0.0.0' all Ipv4 traffic
```
```suggestion
   * Destination Ipv6 address for EGW route
   *
   * @default - '0.0.0.0' all Ipv4 traffic
```

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -1,7 +1,73 @@
+  readonly ipv4Destination?: string;
+
+  /**
+   * Destination Ipv6 address for EGW route
+   * @default '::/0' all Ipv6 traffic
```
```suggestion
   * Destination Ipv6 address for EGW route
   *
   * @default - '::/0' all Ipv6 traffic
```

paulhcsun replied:
Please add a linebreak between the description and any @ default, @ see, @ attribute tags. Also separate the `@default` from the value with a hyphen `-` if you are describing it with a sentence and not the value itself.

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -1,7 +1,73 @@
+  readonly vpnGatewayName?: string;
+
+  /**
+     * Provide an array of subnets where the route propagation should be added.
```
```suggestion
     * Subnets where the route propagation should be added.
```

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -196,6 +323,106 @@ export abstract class VpcV2Base extends Resource implements IVpcV2 {
+      vpc: this,
+    });
+
+    const useIpv6 = (this.secondaryCidrBlock.some((secondaryAddress) => secondaryAddress.amazonProvidedIpv6CidrBlock === true ||
+    secondaryAddress.ipv6IpamPoolId != undefined))? true : false;
```
What is this doing?

shikha372 replied:
Since we can have multiple secondary address attached to VPC, this is to check if any of the secondary address is a IPv6 meaning created using `IPAMIpv6` or is a `AmazonProvidedIpv6`

paulhcsun commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -196,6 +323,106 @@ export abstract class VpcV2Base extends Resource implements IVpcV2 {
+      throw new Error('Egress only IGW can only be added to Ipv6 enabled VPC');
+    };
+
+    if (options?.subnets) {
+      const subnets = flatten(options.subnets.map(s => this.selectSubnets(s).subnets));
+      subnets.forEach((subnet) => {
+        this.createEgressRoute(subnet, egw, options.destination);
```
What is this doing?

shikha372 replied:
this is the default route that we are creating under provided subnet's routeTable. It will add an entry to subnet's routeTable either with the specified destination address or with default(::/0)

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -174,7 +153,18 @@ export interface NatGatewayProps {
+export interface NatGatewayProps extends NatGatewayOptions {
+  /**
+   * The ID of the VPC in which the NAT gateway is located.
+   * Required in case of public connectivity if allocation id is not defined
+   * @default none
```
I wonder why is this a requirement if its not enforced with the linter, also I see it that this is not consistent in whole library, is this something suggested by the core team.
eg. checking this file https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-apigatewayv2/lib/http/authorizer.ts

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -286,7 +291,30 @@ export class VPNGateway extends Resource implements IRouteTarget {
     this.routerTargetId = this.resource.attrVpnGatewayId;
+
     this.vpcId = props.vpc.vpcId;
+    this._attachment = new CfnVPCGatewayAttachment(this, 'VPCVPNGW', {
```
since its a L1, it was suggested in earlier review to store the reference the variable(i suppose which makes it a part of construct tree)

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -174,7 +153,18 @@ export interface NatGatewayProps {
+export interface NatGatewayProps extends NatGatewayOptions {
+  /**
+   * The ID of the VPC in which the NAT gateway is located.
+   * Required in case of public connectivity if allocation id is not defined
```
NATGateway can be `Public` or `Private`, so in case of public connectivity we need allocation id(refers to Elastic IP).

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 5d63c6cf7f5eac960e8c24f30031955ee3df4921
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=094350a4-93ce-48f6-b37e-a777741ad74b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/README.md:
```diff
@@ -195,9 +213,37 @@ const dynamoEndpoint = new ec2.GatewayVpcEndpoint(this, 'DynamoEndpoint', {
+  subnetType: ec2.SubnetType.PRIVATE });
+
+myVpc.addEgressOnlyInternetGateway({
+  subnets: [{SubnetType.PUBLIC}],
```
Haven't looked into implementation but this already looks weird to me.

shikha372 replied:
This is as per the current implementations we have in place for addendpoint and addgateways and provides a better way for the users to select multiple subnets at once without forming an array list first and then pass it. This seems to be the best option for specifying a subnet as an input.

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/awslint.json:
```diff
@@ -2,6 +2,7 @@
       "from-method:@aws-cdk/aws-ec2-alpha.VpcV2",
       "attribute-tag:@aws-cdk/aws-ec2-alpha.RouteTable.routeTableId",
-      "from-method:@aws-cdk/aws-ec2-alpha.SubnetV2"
+      "from-method:@aws-cdk/aws-ec2-alpha.SubnetV2",
+      "from-method:@aws-cdk/aws-ec2-alpha.Route"
```
For my knowledge, what does these two lines do?

shikha372 replied:
this is enforced from `awslint` to have the import methods in classes extending `Resource` , since we haven't introduced it yet these are part of exceptions

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -23,7 +25,7 @@ export enum NatConnectivityType {
  * egress-only internet gateway or VPC endpoint.
  */
-export interface IRouteTarget {
+export interface IRouteTarget extends IDependable{
```
needs a space before `{`

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -71,45 +73,22 @@ export interface InternetGatewayProps {
 /**
  * Properties to define a VPN gateway.
  */
-export interface VPNGatewayProps {
```
This is worrying. This is an exported interface and users might use it for some purposes.

shikha372 replied:
If you mean with that they might be extending it somewhere for implementation, we don't bind the contract of keeping the interface aligned even in the main lib(checked with core team for one of the route interfaces), for the existing VPNGW props this remains same overall, only breaking change that we specify is the change from VPNGateway to VPNGatewayV2, explained below why we need to do that.

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -71,45 +73,22 @@ export interface InternetGatewayProps {
   /**
    * The subnet in which the NAT gateway is located.
    */
-  readonly subnet: ISubnet;
```
Isn't this a breaking change for users?

shikha372 replied:
if mean changing the ISubnet to ISubnetv2, its not breaking change since ISubnetv2 extends ISubnet

GavinZZ reviewed (changes requested):
I'm a bit worried by the number of breaking changes. I understand it is an alpha module and breaking changes are allowed. However, a number of changes are removing an old construct and introduced V2, i.e. `VPNGateway` is removed and introduced `VPNGatewayV2`. It's breaking changes anyway, should we just use the same name?

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/README.md:
```diff
@@ -195,9 +213,37 @@ const dynamoEndpoint = new ec2.GatewayVpcEndpoint(this, 'DynamoEndpoint', {
+VPCv2 supports adding an egress only internet gateway to VPC with the help of `addEgressOnlyInternetGateway` method as well.
+
+By Default, it sets up a route to all outbound IPv6 Address ranges unless specified to a specific destination by the user. It can only be set up for IPv6 enabled VPCs.
+`Subnets` takes in value of `SubnetFilter` which can be based on a SubnetType in VPCV2. A new route will be added to route tables of all subnets filtered out with this property.
```
Selecting a type of subnet is a `SubnetFilter` :) 
`subnets:` [{SubnetType.PUBLIC}]

shikha372 commented:
> Only half way through.
> 
> I'm a bit worried by the number of breaking changes. I understand it is an alpha module and breaking changes are allowed. However, a number of changes are removing an old construct and introduced V2, i.e. `VPNGateway` is removed and introduced `VPNGatewayV2`. It's breaking changes anyway, should we just use the same name?

So one is compatible with `Route` construct and other lives in the main lib `VPNGateway` is not implementing `IRouteTarget` so that can not be set as an target while defining a route. So, the V2 is aligned with features (i.e. `Route` and `RouteTable`) in this new module, it is more clear that this is latest. Also, if there is already a construct with same name in main lib, not sure if its best practice to introduce another with same name as it can create issue with graduation.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 669c735bfcc1b7f51b97d18a8c36b04289202179
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d059c798-028c-4761-83d7-6d9aba0ae644%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/README.md:
````diff
@@ -129,47 +129,65 @@ const subnet = new vpc_v2.SubnetV2(this, 'Subnet', {
 });
 ```
 
+Alternatively, `Route`s can be created via a method in the `RouteTable` class. An example using the `EgressOnlyInternetGateway` construct can be seen below:
````
1. corrected
2. it does relates with creating the route in line , you cannot create a route without a target so we need a target first(in this case), it is EgressOnlyInternetGateway
routeTable.addRoute('::/0', { gateway: eigw });
To make it more clear mentioned the method name

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/README.md:
```diff
@@ -195,9 +213,37 @@ const dynamoEndpoint = new ec2.GatewayVpcEndpoint(this, 'DynamoEndpoint', {
+
+An egress-only internet gateway is a horizontally scaled, redundant, and highly available VPC component that allows outbound communication over IPv6 from instances in your VPC to the internet, and prevents the internet from initiating an IPv6 connection with your instances. For more information see@ https://docs.aws.amazon.com/vpc/latest/userguide/egress-only-internet-gateway.html
+
+VPCv2 supports adding an egress only internet gateway to VPC with the help of `addEgressOnlyInternetGateway` method as well.
```
checked to confirm i think its aligned everywhere else in the Readme, just not in the two places in this section corrected it.

shikha372 replied:
Thanks @GavinZZ , I checked and its only in this block that its `VPCv2` but I think Readme had pending updates for some other add methods, have updated it.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: a9096f55170933c2ec1747b0936652c57479116c
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8b1ef05b-280a-4d33-be62-ff88dc16caa0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -331,6 +358,12 @@ export class NatGateway extends Resource implements IRouteTarget {
     this.connectivityType = props.connectivityType || NatConnectivityType.PUBLIC;
     this.maxDrainDuration = props.maxDrainDuration || Duration.seconds(350);
 
+    if (this.connectivityType == NatConnectivityType.PUBLIC) {
```
nvm, I found that `connectivityType` is a string type.

GavinZZ replied:
should we make it a Enum typed value?

shikha372 replied:
it is an enum typed value in props, so the user pass an enum for this value not string.

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -196,6 +327,106 @@ export abstract class VpcV2Base extends Resource implements IVpcV2 {
+      vpc: this,
+    });
+
+    const useIpv6 = (this.secondaryCidrBlock.some((secondaryAddress) => secondaryAddress.amazonProvidedIpv6CidrBlock === true ||
+    secondaryAddress.ipv6IpamPoolId != undefined))? true : false;
```
`some` already returns a boolean.
```suggestion
    const useIpv6 = this.secondaryCidrBlock.some((secondaryAddress) => secondaryAddress.amazonProvidedIpv6CidrBlock === true ||
    secondaryAddress.ipv6IpamPoolId != undefined);
```

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -408,7 +441,7 @@ export class RouteTargetType {
  * Interface to define a route.
  */
-export interface IRouteV2 {
+export interface IRoute extends IConstruct, IResource{
```
I see some discrepancy where some changes are from v1 to v2 and some changes are from v2 to v1. Can we make it consistent?

shikha372 replied:
i  think its supposed be `IRoute` only, we implemented an idea in between of extending this for introducing one field but went back to the [existing one](https://github.com/aws/aws-cdk/blob/vpcv2-rt-addGateway/packages/%40aws-cdk/aws-ec2-alpha/lib/route.ts#L444C33-L444C54) , this is same as current but i see a nit here `IResource {` that needs to be fixed.
V2 are mainly for the constructs that already exist.

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -1,7 +1,77 @@
+  readonly vpnGatewayName?: string;
+
+  /**
+     * Subnets where the route propagation should be added.
+     * @default noPropagation
+     */
+  readonly vpnRoutePropagation?: SubnetSelection[];
```
yeah, got to know it from @paulhcsun , so working to reformat this for entire repo in next revision,

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -286,7 +291,29 @@ export class VPNGateway extends Resource implements IRouteTarget {
+
+    // Propagate routes on route tables associated with the right subnets
+    const vpnRoutePropagation = props.vpnRoutePropagation ?? [{}];
+    const routeTableIds = allRouteTableIds(flatten(vpnRoutePropagation.map(s => props.vpc.selectSubnets(s).subnets)));
```
`allRouteTableIDs` signifies that there are multiple because of possible multiple subnets, i feel its more clear than getRoutTableIds

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -286,7 +291,29 @@ export class VPNGateway extends Resource implements IRouteTarget {
+    const routeTableIds = allRouteTableIds(flatten(vpnRoutePropagation.map(s => props.vpc.selectSubnets(s).subnets)));
+
+    if (routeTableIds.length === 0) {
+      Annotations.of(this).addError(`enableVpnGateway: no subnets matching selection: '${JSON.stringify(vpnRoutePropagation)}'. Select other subnets to add routes to.`);
```
since its optional, i'll modify it be a warning.

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -462,7 +495,7 @@ export interface RouteProps {
  * @resource AWS::EC2::Route
  */
-export class Route extends Resource implements IRouteV2 {
+export class Route extends Resource implements IRoute, IDependable {
```
This helps in specifying dependencies within construct. 
So this was needed in order to create a dependency on the gateways and endpoints creation before creation of route. It should start only after your target is created else the deployment will fails. This is handled as below in code.

```
    if (this.target.gateway) {
      this.node.addDependency(this.target.gateway);
    }
    if (this.target.endpoint) {
      this.node.addDependency(this.target.endpoint);
    }
```

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/lib/subnet-v2.ts:
```diff
@@ -250,6 +257,14 @@ export class SubnetV2 extends Resource implements ISubnetV2 {
+  /**
+   * Return the Route Table associated with this subnet
+   */
+  public get routeTable(): IRouteTable {
```
so this was tied to implementation of `[associateRouteTable](https://github.com/aws/aws-cdk/pull/31224/commits/f904a97031a301787b16db7f14619b4e67868a7e#diff-fe8bcfc72ac04e9cea500b3fd101b842e2bb77a648a3e6d0c7fdd2d3cdb16ba0R258)` method in SubnetV2, which we also reverted but we can introduce it in the next release with imports so prefer to keep it as is now currently unless you see any blocking concerns.

shikha372 commented on packages/@aws-cdk/aws-ec2-alpha/lib/subnet-v2.ts:
```diff
@@ -145,19 +153,16 @@ export class SubnetV2 extends Resource implements ISubnetV2 {
    * @attribute SubnetType
    */
-  public readonly subnetType: SubnetType;
+  public readonly subnetType?: SubnetType;
```
This is not optional under props, so user still needs to defined it but it is marked as optional under the new interface `SubnetV2` because if not it will cause issue with the existing property in [`SubnetSelection` for subnets ](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-ec2/lib/vpc.ts#L363) which we are leveraging in some of the places [like here.](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-ec2-alpha/lib/subnet-v2.ts#L351)

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 5c7548c4954c8b3433cef4b45e7f0955825ddf1b
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c7714e15-203c-4920-9254-eaaa055e567a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/route.ts:
```diff
@@ -331,9 +369,15 @@ export class NatGateway extends Resource implements IRouteTarget {
     // If user does not provide EIP, generate one for them
     var aId: string | undefined;
-    if (this.connectivityType == 'public') {
+    if (this.connectivityType == NatConnectivityType.PUBLIC) {
```
`===`

shikha372 replied:
i see it updated in the last revision, not sure if you;re still seeing the same https://github.com/aws/aws-cdk/pull/31224/files/5c7548c4954c8b3433cef4b45e7f0955825ddf1b#diff-65117dd44ab7b3f0b0f8d0421973a71b104cd76b4dc5147237da60616c718b2fR372

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -1,7 +1,80 @@
+/**
+ * Options to define EgressOnlyInternetGateway for VPC
+ */
+export interface EgressOnlyInternetGatewayOptions{
```
```suggestion
export interface EgressOnlyInternetGatewayOptions {
```

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -196,6 +333,108 @@ export abstract class VpcV2Base extends Resource implements IVpcV2 {
+    this._internetConnectivityEstablished.add(igw);
+    this._internetGatewayId = igw.routerTargetId;
+
+    //If there are no public subnets defined, no default route will be added
```
```suggestion
    // If there are no public subnets defined, no default route will be added
```

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/lib/vpc-v2-base.ts:
```diff
@@ -196,6 +333,108 @@ export abstract class VpcV2Base extends Resource implements IVpcV2 {
+      throw new Error('No public subnets defined to add route for internet gateway');
+    }
+
+    //Add deffault route to IGW for IPv6
```
```suggestion
    // Add default route to IGW for IPv6
```

GavinZZ commented on packages/@aws-cdk/aws-ec2-alpha/test/integ.ipam.ts:
```diff
@@ -0,0 +1,84 @@
+// /*
+//  * Our integration tests act as snapshot tests to make sure the rendered template is stable.
+//  * If any changes to the result are required,
+//  * you need to perform an actual CloudFormation deployment of this application,
+//  * and, if it is successful, a new snapshot will be written out.
+//  *
+//  * For more information on CDK integ tests,
+//  * see the main CONTRIBUTING.md file.
+//  */
```
duplicate `//`

GavinZZ reviewed (dismissed):
Thanks Shikha for this great work. The code generally looks good to me. I have two high-level thoughts:
1. I know I've mentioned this, but seeing the usage and it's in `alpha` modules, I'm not convinced why we need to remove the original constructs, i.e. `VPNGateway, Subnet` and create a `VPNGatewayV2, SubnetV2`. Either way, it's breaking changes for customers, wouldn't keep using the name `VPNGateway and Subnet` but with new implementation be better if in the future we decide to make it stable and it won't be starting from `V2` by default.
2. Subnets have to be attached with a VPC resource. Can we introduces methods in VPC construct like `vpc.addSubnet(...)` just like how you added `vpc.addNatGateway, vpc.addInternetGateway`. I peronsally think it makes sense to add subnet this way and prefer creating subnets this way instead of `new SubnetV2`. Want to see your thoughts.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 4b4fb6b324a8b2d6c2a52f1d203db091bf846808
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5c7ecb1f-ae36-4ecf-83b4-75bd61c5d9a6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 8dffdbecf74eac428461bad9c681a50cec34c487
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6ba2d3bc-6a69-4f5b-8e81-7c565f4bb2f6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 753273f03975cf3793a1a8134a838d50eb039cab
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=58a422fe-a9a2-46d6-b705-bc1f79cc692f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

paulhcsun reviewed (approved):
Looks good to me, deferring final approval to @GavinZZ. Nice work @shikha372!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: dcb11e636046b918177d4a0ea62aa48050f30868
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8e957cb9-7a00-4bd2-9231-7046659636fd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

shikha372 commented:
@mergifyio update

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: f30e1da51cdbc8a127b9f09737c2620f9552a7e8
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=878bfaee-b7a8-46b2-aed2-2036f0be79b4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
